src/TraceEvent/TraceEventSession.cs

```diff
@@ -1368,6 +1368,35 @@ public static unsafe List<string> GetActiveSessionNames()
             return activeTraceNames;
         }
 
+        /// <summary>
+        /// Get the maximum number of ETW loggers supported by the current machine
+        /// </summary>
+        /// <returns>The maximum number of supported ETW loggers</returns>
+        private static int GetETWMaxLoggers()
```

Comment:
Can you cache the value in a static field?  When the registry key changes, the machine requires a reboot to apply the change, so it is safe to cache this for the life of the process.

Reply:
Good call. I have cached the value. 
I also added validation on permissible values defined https://docs.microsoft.com/en-us/windows/win32/api/evntrace/nf-evntrace-starttracew